lib/util/meta.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+var META_SYMBOL = Symbol('json-schema-ref-parser meta');
+
+exports.META_SYMBOL = META_SYMBOL;
+
+/**
+ * Safely sets meta property on the given object value
+ *
+ * @param  {object} obj - A object to set meta info on
+ * @param  {string} prop - Name of property
+ * @param  {string} prop - Value to set
+ */
+exports.setMeta = function (obj, prop, value) {
+  if (!obj) { return; }
+  // we can't create property on primitives (throws in strict mode)
+  if (typeof obj !== 'object') { return; }
+
+  if (!obj[META_SYMBOL]) {
+    obj[META_SYMBOL] = {};
+  }
+  obj[META_SYMBOL][prop] = value;
+};
+
+/**
+ * Given the any object returns the associated meta-information or undefined
+ *
+ * @param  {any} value - object to get the meta from
+ * @returns {{pointer: string}}
+ */
+exports.getMeta = function (value) {
```

Comment:
Let's rename `getMeta()` to `getMetadata()`. It's more explicit/clear.

```suggestion
exports.getMetadata = function (value) {
```